appservice/src/connectToGitHub.ts

```diff
@@ -14,76 +13,72 @@ import { ext } from './extensionVariables';
 import { localize } from './localize';
 import { signRequest } from './signRequest';
 import { SiteClient } from './SiteClient';
+import { ISiteTreeRoot } from './tree/ISiteTreeRoot';
 import { nonNullProp } from './utils/nonNull';
 
 type gitHubOrgData = { repos_url?: string };
 type gitHubReposData = { repos_url?: string, url?: string, html_url?: string };
 type gitHubLink = { prev?: string, next?: string, last?: string, first?: string };
 // tslint:disable-next-line:no-reserved-keywords
 type gitHubWebResource = WebResource & { resolveWithFullResponse?: boolean, nextLink?: string, lastLink?: string, type?: string };
+const badCredentials: string = 'Bad credentials';
 
-export async function connectToGitHub(node: AzureTreeItem, client: SiteClient): Promise<void> {
+export async function connectToGitHub(ti: AzureTreeItem<ISiteTreeRoot>): Promise<void> {
+    const client: SiteClient = ti.root.client;
+    let siteSourceControl: SiteSourceControl;
     const requestOptions: gitHubWebResource = new WebResource();
-    let repoSelected: boolean = false;
     requestOptions.resolveWithFullResponse = true;
     requestOptions.headers = {
         ['User-Agent']: 'vscode-azureappservice-extension'
     };
     const oAuth2Token: string | undefined = (await client.listSourceControls())[0].token;
     if (!oAuth2Token) {
-        await showGitHubAuthPrompt();
+        // we pass this error to validate the error was due to be bad cred and give the user an actionable error
+        await showGitHubAuthPrompt(ti, new Error(badCredentials));
```

Comment:
Sorry, I'm not entirely sure how this would resolve the issue.  There is no error message for having no OAuth2 token because it just is undefined which is why I throw the created error.  That is the same error that GitHub rejects with if your credentials have expired.  The actionable item is the same for both scenarios which is why I handle it with the `showGitHubAUthPrompt`